### ISSUE TITLE
gatsby-plugin-sharp 2.0.0-alpha.19 and/or gatsby-transformer-sharp 2.0.1-12 are broken

### ISSUE THREAD

**Issue by tsiq-swyx** [gatsbyjs/gatsby#5690]
## Description

```
Error: Invariant Violation: GraphQLCompilerContext: Unknown document `GatsbyImageSharpResolutions_tracedSVG`.

  - invariant.js:42 invariant
    [www]/[fbjs]/lib/invariant.js:42:15

  - GraphQLCompilerContext.js:145 GraphQLCompilerContext._get
    [www]/[relay-compiler]/lib/GraphQLCompilerContext.js:145:34

  - GraphQLCompilerContext.js:116 GraphQLCompilerContext.getFragment
    [www]/[relay-compiler]/lib/GraphQLCompilerContext.js:116:21

  - filterContextForNode.js:29 Object.FragmentSpread
    [www]/[relay-compiler]/lib/filterContextForNode.js:29:32

# and so on...
```

when running `/www`@v2 branch.

### Steps to reproduce

```
git checkout v2
yarn
yarn run develop
```

### Expected result

the docs site should build successfully

### Actual result

![image](https://user-images.githubusercontent.com/35976578/40892754-d3bbea6c-6768-11e8-98c3-ff1c382e2487.png)

### Environment
n/a

### File contents (if changed)
n/a

---

# known solution:

pin versions of 

```
"gatsby-plugin-sharp": "2.0.0-alpha.16",
"gatsby-transformer-sharp": "2.0.1-9",
```

instead of the `@next` version and it builds fine. so something in 2.0.0-alpha.19 or 2.0.1-12 broke.

**Comment by tsiq-swyx** [comment]
It looks like @KyleAMathews pushed something on friday to both libraries.

**Comment by pieh** [comment]
I just checked with clean repo and this does work both locally and on netlify - https://app.netlify.com/sites/gatsbyjs/deploys/5b151e0c11b73b62b621109d, so this doesn't seem like problem with packages themselves but maybe with gatsby cache here? Can you still reproduce it?

**Comment by pieh** [comment]
Ok, now I understand why I couldn't reproduce - i was using fresh master which uses new fragments - you probably have older master with older fragments names, but installed fresh deps. I will open PR to restore old fragments - same thing @ryanditjia did for contentful plugin https://github.com/gatsbyjs/gatsby/pull/5678

**Comment by m-allanson** [comment]
Should be fixed by #5710, thanks @pieh!

**Comment by sw-yx** [comment]
ok @pieh thanks I don't actually understand where the fragments are injected, that's honestly some of the magic of Gatsby that could be a lot better. we need to think about how to make these things debuggable.
